### PR TITLE
Added flushUntil prop to Section

### PR DESCRIPTION
**Pull request by mperrotti**
#### Related issues
Fixes https://meetup.atlassian.net/browse/SDS-302

#### Description
`<Section>` can now be flushed left and right. Supports media-conditional flushing

### CODE REVIEW

File: src/layout/Section.jsx [hunk 1]
```diff
@@ -47,6 +51,10 @@ Section.propTypes = {
 	noSeparator: PropTypes.oneOfType([
 		PropTypes.bool,
 		PropTypes.oneOf(Object.keys(VALID_BREAKPOINTS))
+	]),
+	flushUntil: PropTypes.oneOfType([
+		PropTypes.bool,
```
Comment (akdetrick) [inline src/layout/Section.jsx]: feels weird to support a boolean because it's not immediately clear what this means:

`<Section flushUntil>...`

`<Section className="flush--horizontal">` may be more useful for the use case of flushing at all breakpoints. This doesn't exist in sasstools yet, but it's an easy addition...

Comment (mperrotti) [inline src/layout/Section.jsx]: Makes sense to me. I wanted to add a way to _always_ flush a `<Section>`, but I guess it's rare enough that we could just do it in `sasstools`

File: src/layout/Section.jsx [hunk 2]
```diff
@@ -47,6 +51,9 @@ Section.propTypes = {
 	noSeparator: PropTypes.oneOfType([
 		PropTypes.bool,
 		PropTypes.oneOf(Object.keys(VALID_BREAKPOINTS))
+	]),
+	flushUntil: PropTypes.oneOfType([
```
Comment (nlaz) [inline src/layout/Section.jsx]: Since you got rid of the bool proptype, can't you shorten this to something like:`flushUntil: PropTypes.oneOf(Object.keys(VALID_BREAKPOINTS)),`?

Comment (mperrotti) [inline src/layout/Section.jsx]: Good idea. On it...

**Comment by akdetrick** [comment]
My `flushUntil="all"` nitpick shouldn't block this merge; approving.
